pylint/checkers/base.py

```diff
@@ -1872,6 +1894,11 @@ def visit_unaryop(self, node):
             operator, right = operand.ops[0]
             if operator not in self.reverse_op:
                 return
+            for _type in (_node_type(left), _node_type(right)):
+                if not _type:
+                    return
+                if isinstance(_type, astroid.Set):
```

Comment:
Regarding the second point: if the set can't be inferred, then _node_type() will return None, and the function will return without emitting a message. 
Or maybe I didn't understand what you said?